### PR TITLE
feat(typescript): improve docs generation

### PR DESCRIPTION
**Pull request by joscha**
This improves the docs for generated typescript code a bit:

Screenshot 2024-09-27 at 2 00 57 AM

The [split diff](https://github.com/OpenAPITools/openapi-generator/pull/19699/files?diff=split&w=0) is way more readable for this pull request than the unified diff.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language,  @macjohnny

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript/api_doc.mustache [hunk 1]
```diff
@@ -22,27 +22,25 @@ Method | HTTP request | Description
 
 
 ```typescript
-import { {{{moduleName}}} } from '{{{projectName}}}';
-import * as fs from 'fs';
+import {{{moduleName}}} from '{{{projectName}}}';
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api_doc.mustache:25]: we're using a default export here, alternatively we could do:
```suggestion
import { {{classname}}, createConfiguration } from '{{{projectName}}}';
```
if we want to keep the non-default export.

File: modules/openapi-generator/src/main/resources/typescript/api_doc.mustache [hunk 2]
```diff
@@ -22,27 +22,25 @@ Method | HTTP request | Description
 
 
 ```typescript
-import { {{{moduleName}}} } from '{{{projectName}}}';
-import * as fs from 'fs';
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api_doc.mustache:26]: not sure when and why the `fs` module was added.

File: modules/openapi-generator/src/main/resources/typescript/api_doc.mustache [hunk 3]
```diff
@@ -22,27 +22,25 @@ Method | HTTP request | Description
...
+import {{{moduleName}}} from '{{{projectName}}}';
 
 const configuration = {{{moduleName}}}.createConfiguration();
 const apiInstance = new {{{moduleName}}}.{{classname}}(configuration);
 
 {{#hasParams}}
-let body:{{{moduleName}}}.{{classname}}{{operationIdCamelCase}}Request = {
+const request: {{{moduleName}}}.{{classname}}{{operationIdCamelCase}}Request = {
 {{#allParams}}
-  // {{{dataType}}}{{#description}} | {{{description}}}{{/description}}{{^required}} (optional){{/required}}
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api_doc.mustache:34]: given that we define the request with an explicit type, there is no point adding the dataType into the generated code. Anyone copy & pasting the code into their Typescript project would see what types they can pass.

File: modules/openapi-generator/src/main/resources/typescript/api_doc.mustache [hunk 4]
```diff
@@ -22,27 +22,25 @@ Method | HTTP request | Description
...
 {{/hasParams}}
 {{^hasParams}}
-let body:any = {};
+const request = {};
 {{/hasParams}}
 
-apiInstance.{{{operationId}}}(body).then((data:any) => {
-  console.log('API called successfully. Returned data: ' + data);
-}).catch((error:any) => console.error(error));
+const data = await apiInstance.{{{operationId}}}(request);
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api_doc.mustache:42]: changed this to `await`, then we don't need to do an error handling example and the code gets a bit shorter

File: samples/client/others/typescript/builds/array-of-lists/DefaultApi.md [hunk 5]
```diff
@@ -15,17 +15,15 @@ Method | HTTP request | Description
 
 
 ```typescript
-import {  } from '';
```
Comment (joscha) [inline samples/client/others/typescript/builds/array-of-lists/DefaultApi.md:18]: most of these samples do not define a `moduleName` or `projectName` in their yaml. Examples are also sparse. The petstore is pretty decent.

**Review by macjohnny — APPROVED** [review]
thanks!

**Pull request merged**